src/Compilers/Server/VBCSCompilerTests/DesktopBuildClientTests.cs

```diff
@@ -22,23 +22,32 @@ public abstract class DesktopBuildClientTests : TestBase
     {
         private sealed class TestableDesktopBuildClient : DesktopBuildClient
         {
+            /// <summary>
+            /// Stores fully constructed pipe name.
+            /// </summary>
             private readonly string _pipeName;
+            /// <summary>
+            /// Stores unhashed "uniqueifying" pipe name component.
```

Comment:
We are of course willing to pursue either path. Answering your question directly, however, @phkelley has suggested, in an aforementioned email thread, several scenarios where using sharedCompilationId as a component would be advantageous:

> - We preserve whatever value is derived from the existing automatic pipe name generation logic, and uniquify on top of that
>  - Server is unique to the Windows user
>  - Server is unique to the user’s elevation state
>  - Server is unique to the full path to VBCSCompiler.exe
> - Protection from any illegal characters in the provided string
>  - It will be used as both a mutex name and a pipe name
>  - Backslash, spaces, etc. all handled transparently